### PR TITLE
[WIP]upgrade prometheus-common v0.39.0

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Pending on issue below:
```
go: downloading cloud.google.com/go/compute/metadata v0.2.0
k8s.io/apiextensions-apiserver/pkg/apiserver imports
	k8s.io/apiserver/pkg/registry/generic imports
	k8s.io/apiserver/pkg/storage/storagebackend/factory imports
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc tested by
	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.test imports
	google.golang.org/grpc/interop imports
	golang.org/x/oauth2/google imports
	cloud.google.com/go/compute/metadata: ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules:
	cloud.google.com/go v0.97.0 (/home/prow/go/pkg/mod/cloud.google.com/go@v0.97.0/compute/metadata)
	cloud.google.com/go/compute/metadata v0.2.0 (/home/prow/go/pkg/mod/cloud.google.com/go/compute/metadata@v0.2.0)
Removing /home/prow/go/src/k8s.io/kubernetes/_tmp/kube-vendor.SMrfcB
```

#### Which issue(s) this PR fixes:
Fixes None

#### Special notes for your reviewer:
https://github.com/prometheus/common/releases

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by pacoxu** [comment]
/assign @liggitt @sttts

```
 Direct Dependencies: 211 
-Transitive Dependencies: 234 
-Total Dependencies: 285 
+Transitive Dependencies: 237 
+Total Dependencies: 288 
 Max Depth Of Dependencies: 23 
 All dependencies:
 
@@ -8,6 +8,7 @@
 bitbucket.org/bertimus9/systemstat
 cloud.google.com/go
 cloud.google.com/go/bigquery
+cloud.google.com/go/compute/metadata
 cloud.google.com/go/storage
 github.com/Azure/azure-sdk-for-go
 github.com/Azure/go-ansiterm
@@ -29,6 +30,8 @@
 github.com/NYTimes/gziphandler
 github.com/PuerkitoBio/purell
 github.com/PuerkitoBio/urlesc
+github.com/alecthomas/template
+github.com/alecthomas/units
 github.com/antihax/optional
 github.com/antlr/antlr4/runtime/Go/antlr
 github.com/armon/circbuf
```

File: go.mod [hunk 1]
```diff
@@ -78,7 +78,7 @@ require (
 	go.uber.org/zap v1.19.0
 	golang.org/x/crypto v0.1.0
 	golang.org/x/net v0.4.0
-	golang.org/x/oauth2 v0.0.0-20220223155221-ee480838109b
+	golang.org/x/oauth2 v0.3.0
```
Comment (liggitt) [inline go.mod:81]: /hold

this bump gets us into the middle of the cloud/go/compute/metadata bump and triggers publishing bot breakage
cc @dims @nikhita 

xref https://github.com/kubernetes/kubernetes/issues/113366

**Comment by liggitt** [comment]
/test pull-kubernetes-dependencies
to run check added in https://github.com/kubernetes/kubernetes/pull/114952

**Comment by dgrisonnet** [comment]
No concerns from SIG Instrumentation.

**Pull request closed without merging**